### PR TITLE
INTERLOK-3460 xpath service attribute node support

### PR DESCRIPTION
**Pull request by Co-Din**
## Motivation

Why am I doing this

I discovered that when retrieving a single attribute node in the XPath Service the service would simply throw an exception rather than return the value

## Modification

What did I change

I updated the Xpath service to resolve attribute nodes and updated the service's tests to reflect this.

## Result

What's the end result for the user

The user can now resolve single attribute nodes with the Xpath service

## Testing

How can I test this if I'm reviewing this.

The tests have been updated and cover the changes.
You can also build the project, drop the jar into a working adapter and query an xpath attribute node to confirm

### CODE REVIEW

File: interlok-core/src/main/java/com/adaptris/core/services/path/XPathService.java [hunk 1]
```diff
@@ -248,11 +248,17 @@ private String serializeNode(NodeList nodeList) throws TransformerException {
...
+	    StringWriter stringWriter = new StringWriter();
+		StreamResult xmlOutput = new StreamResult(stringWriter);
+	    Transformer transformer = TransformerFactory.newInstance().newTransformer();
+	    transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+	    if (source.getNode().getNodeType() != Node.ATTRIBUTE_NODE) {
+	        transformer.transform(source, xmlOutput);
+	      } else {
+	        stringWriter.write(source.getNode().getNodeValue());
+	      }
+	    return stringWriter.toString();
```
Comment (quotidian-ennui) [inline interlok-core/src/main/java/com/adaptris/core/services/path/XPathService.java:261]: Spaces not tabs... (when I view the while file, this sticks out like a sore thumb).

**Comment by quotidian-ennui** [comment]
What's the "point" of the change; I mean I understand the point of the change, but why has it manifested itself?

- Do we get the problem with XPathMetadataService if we run the xpath + xml against that?
- Is it just XpathService because it does selectNodeList and then tries to render it that as text?

- Could you write a test for XpathMetadataService that does the same test as this (that's the one that most people will be using).

File: interlok-core/src/main/java/com/adaptris/core/services/path/XPathService.java (hunk 1 above)
Comment (Co-Din) [reply interlok-core/src/main/java/com/adaptris/core/services/path/XPathService.java]: Apologies, just updated the indentations to two spaces

**Comment by Co-Din** [comment]
@quotidian-ennui 

Manifested while trying to create an example using the Xpath service for documentation, the service couldn't resolve attribute nodes.

We don't get the same problem with XpathMetadata, in fact we can find and will resolve all values matching the node attribute.

 Is it just XpathService because it does selectNodeList and then tries to render it that as text?
It seems when I followed in debug that in XpathMetadata when it's configuring the Xpath Metadata query it's actually setting it to a MultiItemConfiguredXpathQuery rather than ConfiguredXpathQuery though I'm haven't managed to discern where this is happening

Could you write a test for XpathMetadataService that does the same test as this (that's the one that most people will be using).

Now added two tests for single and multiple matches to attribute node queries

**Review by quotidian-ennui — APPROVED** [review]

**Pull request merged**